CenterForOpenScience/ember-osf-web
# [ENG-4113] Collections Moderator Management

-   Ticket: [ENG-4113]
-   Feature flag: n/a

## Purpose
- Allow moderator management for collection-providers

## Summary of Changes
- Prevent issue on mobile view where provider-logo could flow out of header container on discover page
- Move moderator logic to provider-superclass
  - Move `moderators` relationship
  - Move `permissions` attribute
  - Move `currentUserCanReview` logic
- Check if the currentUser is actually a moderator to show "Moderation" button on navbar
- Tweak Moderator manager component to be less tied to registries implementation
- Use Moderator manager for collection-provider
- Tests + test selectors

## Screenshot(s)
- Admin view:
![image](https://user-images.githubusercontent.com/51409893/198387886-8384df7d-ea53-40a0-adb8-56dbe60047a9.png)
- Moderator view:
![image](https://user-images.githubusercontent.com/51409893/198389036-f48d9639-276f-494a-9fae-358640c8f5ec.png)


## Side Effects
- I don't think there will be any side-effects... 🤔 

## QA Notes
- Admins should be able to add moderators by user_id or by email
- Non-admin moderators should only be able to remove themselves

[ENG-4113]: https://openscience.atlassian.net/browse/ENG-4113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
